File: cpp/RT_CPP/RT_CPP/texture_material.cpp
#include "texture_material.h"

using namespace materials;

texture_material::texture_material(int mid, material_types tp, vector<vector3> acs,
	vector<texture*> dcs, vector<vector3> spcs, vector<float> specs,
	vector<float> rfs, vector<float> rts, vector<float> rfIs) : material(mid, tp, specs, rfs, rts, rfIs),
	ambient_colors(acs), diffuse_colors(dcs), specular_colors(spcs)
{

}

vector3 texture_material::get_ambient_color(vector2 uv) const
{
	return ambient_colors[0];
}

vector3 texture_material::get_diffuse_color(vector2 uv) const
{
	return diffuse_colors[0]->get_pixel(uv);
}
vector3 texture_material::get_specular_color(vector2 uv) const
{
	return specular_colors[0];
}
float texture_material::get_specularity(vector2 uv) const
{
	return specularities[0];
}

float texture_material::get_reflectivity(vector2 uv) const
{
	return reflectivities[0];
}

float texture_material::get_transparency(vector2 uv) const
{
	return transparencies[0];
}

float texture_material::get_refractive_index(vector2 uv) const
{
	return refractiveIndices[0];
}

File: Csharp/RT_CS/RT_CS/Materials/TextureMaterial.cs
using RT_CS.utility;
using System;
using System.Collections.Generic;
using System.Text;

namespace RT_CS.Materials
{
    public class TextureMaterial : Material
    {

        private List<Vector3> ambientColors, specularColors;
        private List<Texture> diffuseColors;

        public TextureMaterial(int mid, List<float> spcs, MaterialType mt, List<float> refl, List<float> trans, List<float> refr,
            List<Vector3> ac, List<Texture> dc, List<Vector3> sc) : base(mid, spcs, mt, refl, trans, refr)
        {
            ambientColors = ac;
            diffuseColors = dc;
            specularColors = sc;

        }

        public override Vector3 GetAmbientColor(Vector2 uv)
        {
            return ambientColors[0];
        }

        public override Vector3 GetDiffuseColor(Vector2 uv)
        {
            return diffuseColors[0].GetPixel(uv);
        }

        public override Vector3 GetSpecularColor(Vector2 uv)
        {
            return specularColors[0];
        }

        public override float GetSpecularity(Vector2 uv)
        {
            return specularities[0];
        }

        public override float GetReflectivity(Vector2 uv)
        {
            return reflectivity[0];
        }
        public override float GetTransparency(Vector2 uv)
        {
            return transparency[0];
        }
        public override float GetRefractiveIndex(Vector2 uv)
        {
            return refractiveIndices[0];
        }
    }
}
